nixos/tests/mailman.nix

```diff
@@ -63,5 +63,11 @@ import ./make-test-python.nix {
         wait_for_api()
         machine.succeed("curl --fail-with-body -sLSu restadmin:secretpassword http://localhost:8001/3.1/domains")
         machine.succeed("curl --fail-with-body -sILS http://localhost/")
+
+    with subtest("service locking"):
+        machine.fail("su -s /bin/sh -c 'mailman start' mailman")
+        machine.execute("systemctl kill --signal=SIGKILL mailman")
```

Comment:
Interesting. Just for clarification: the systemd management means that the unit failure (and cgroup cleanup) causes a cleanup of the run directory, too? I was a bit surprised when reading this test as I was expecting we'd have to actually crash the machine ...

Reply:
From [systemd.exec(5)](https://www.freedesktop.org/software/systemd/man/systemd.exec.html): "In case of `RuntimeDirectory=` the innermost subdirectories are removed when the unit is stopped." When mailman stops, systemd will by default automatically remove `/run/mailman` and its contents, which with this change includes the lockfiles.

Reply:
Thought so, thanks! That makes a lot of sense and gives us robustness in both crashes and reboots. Nice.